packages/babel-core/src/config/config-descriptors.ts

```diff
@@ -133,22 +133,26 @@ export function createUncachedDescriptors(
     options: optionsWithResolvedBrowserslistConfigFile(options, dirname),
     *plugins() {
       if (!plugins) {
-        plugins = yield* createPluginDescriptors(
+        const _plugins = yield* createPluginDescriptors(
           options.plugins || [],
           dirname,
           alias,
         );
+        // check plugins again to avoid race conditions
+        plugins ??= _plugins;
```

Comment:
Instead of just ignoring the result of the second `createPluginDescriptors` call, can we avoid calling it twice in the first place? Something like this:
```js
  let pluginsGen;
  return {
    options: optionsWithResolvedBrowserslistConfigFile(options, dirname),
    *plugins() {
      pluginsGen ??= createPluginDescriptors(
        options.plugins || [],
        dirname,
        alias,
      );
      return yield* pluginsGen;
    }
  };
```

Reply:
If we cache the iterator instead, test will fail because the finished iterator always return the last passed `.next()` argument, which in this case, is always `undefined`: https://github.com/JLHwung/babel/runs/7819160996?check_suite_focus=true

Reply:
I pushed a commit to do the caching, so that now:
- The first call runs `createPluginDescriptors`, and after a while returns it
- The second call waits for the result of the first call, and returns it

Instead of
- The first call runs `createPluginDescriptors`, and returns the result of the `createPluginDescriptors` that finishes first
- The second call runs `createPluginDescriptors`, and returns the result of the `createPluginDescriptors` that finishes first

This avoids resolving the same plugins/presets multiple times, which is a costly operation because it involves the filesystem.